Otavio/feat/create user auth pages

otavioadamis opened the pull request:
This PR updates user authentication in both frontend and backend.

Frontend:
- Implements sign-in and sign-up pages according to the Figma design.
- Adds reset password flow: request, 5-digit email verification code, and new password form.
- Additionally, updates email confirmation page to match the app’s design.

Backend:
- Adds custom login and register serializers so API requests match frontend forms.
- Introduces models to store password reset tokens and 5-digit email verification codes, linked to the requesting user.
- Implements serializers and views for creating and verifying the 5-digit email code and password reset token.

closes #1
closes #3

Images:

<img width="384" height="841" alt="image" src="https://github.com/user-attachments/assets/01375268-e4cb-465c-9af2-a11d638f831a" />
<img width="390" height="838" alt="image" src="https://github.com/user-attachments/assets/9ff053e3-6a75-41dd-9bd9-9f44a7866cc1" />
<img width="381" height="831" alt="image" src="https://github.com/user-attachments/assets/d7c09782-c2aa-40b2-847f-fb8fc8c9e404" />
<img width="380" height="834" alt="image" src="https://github.com/user-attachments/assets/ff387bfc-5f6c-412b-a63c-f1f2de4700a9" />
<img width="382" height="839" alt="image" src="https://github.com/user-attachments/assets/fba449ed-28af-4029-a222-745dd1f88bb4" />
<img width="388" height="837" alt="image" src="https://github.com/user-attachments/assets/d489dec1-01a5-4449-841e-e94bbd6a8f56" />
<img width="386" height="840" alt="image" src="https://github.com/user-attachments/assets/1cf169cf-7240-4aeb-ac82-acedbcd6462c" />

z-sega commented:
Excellent work @otavioadamis!
Thanks for taking on the backend aspects too, very helpful. 

Recommendations:
- Form Validation should take advantage of the backend response when things go wrong, say during sign-in the credentials are bad, the authentication backend sends the following:
<img width="742" height="76" alt="image" src="https://github.com/user-attachments/assets/b1119fe9-242d-4f0d-85de-a7b0fc82830e" />
The frontend should display these as form validation errors. A common pattern is to have room for field specific errors (e.g passwords not matching) and non-field errors (like in the screenshot above: these are typically displayed above the form or above the forms action button).

z-sega reviewed (changes requested):
Great work! 

Please check out my comment on the matter of form validation.

otavioadamis commented:
> Great work!
> 
> Please check out my comment on the matter of form validation.

@z-sega Thanks! 🙌

I’ve refactored the form validation by creating a generic parseBackendErrors helper, so field errors are mapped correctly to their respective form fields and non field errors appears above the action button. I also added a fallback message in cases the API is unreachable.

See examples of backend responses handled:

Log in with invalid credentials:
<img width="388" height="839" alt="image" src="https://github.com/user-attachments/assets/67047bba-bd26-4e15-8411-2fc528aa1212" />

Sign up with already registered email:
<img width="390" height="830" alt="image" src="https://github.com/user-attachments/assets/42cfb6c2-4232-4df2-84ff-8fac18d1868d" />
